tests/ft_puppet.vader

```diff
@@ -0,0 +1,22 @@
+Include: include/setup.vader
+
+Execute (puppet parser, syntax error: end of input):
+  new
+  let maker = NeomakeTestsFixtureMaker('neomake#makers#ft#puppet#puppet',
```

Comment:
You should use `file tests/fixtures/input/puppet/syntax-error-eoi.pp` - otherwise errors are not generated for the current buffer, but a new one.

Reply:
Not sure though - maybe `edit/split tests/fixtures/input/puppet/syntax-error-eoi.pp` (possibly prefixed with `noautocmd`) would be better?!

Reply:
It uses a tempfile anyway - would need to check this myself:

> Using tempfile for unnamed buffer: "/tmp/nvimLiCMEo/112/neomaketmp.".

You can use `Log neomake#utils#redir('ls!')` to get a list of buffers to see what's going on.

Reply:
> Not sure though - maybe edit/split tests/fixtures/input/puppet/syntax-error-eoi.pp (possibly prefixed with noautocmd) would be better?!

I don't understand it, could you clarify?

Reply:
It only uses `new` to create an unnamed buffer, but the errors are for another file - therefore the assertion in the test fails.  (Vim creates an unlisted buffer when parsing the output)